core/src/components/datetime/datetime.tsx

```diff
@@ -2071,20 +2079,39 @@ export class Datetime implements ComponentInterface {
       </ion-popover>,
     ];
   }
-  private renderCalendarViewHeader(mode: Mode) {
+  private renderCalendarViewHeader() {
     const hasSlottedTitle = this.el.querySelector('[slot="title"]') !== null;
     if (!hasSlottedTitle && !this.showDefaultTitle) {
       return;
     }
 
+    const { activeParts, selectedDatesLabelFormatter } = this;
+    const isArray = Array.isArray(activeParts);
+
+    let headerText;
+    if (isArray && activeParts.length !== 1) {
+      const defaultText = `${activeParts.length} days`;
```

Comment:
Can we just the `headerText` here, so we don't need to have additional assignments for exceptions/else checks?

e.g.:
```ts
headerText = `${activeParts.length} days`;
if (selectedDatesLabelFormatter !== undefined) {
  try {
    headerText = selectedDatesLabelFormatter(activeParts.length);
  } catch (e) {
    printIonError('Exception in provided `selectedDatesLabelFormatter`: ', e);
  }
}
```